webcrypto/aes.go

```diff
@@ -0,0 +1,84 @@
+package webcrypto
+
+import (
+	"crypto/rand"
+)
+
+// AesKeyGenParams represents the object that should be passed as
+// the algorithm parameter into `SubtleCrypto.generateKey`, when generating
+// an AES key: that is, when the algorithm is identified as any
+// of AES-CBC, AES-CTR, AES-GCM, or AES-KW.
+//
+// [specification]: https://www.w3.org/TR/WebCryptoAPI/#aes-keygen-params
+type AesKeyGenParams struct {
+	Algorithm
+
+	// The length, in bits, of the key.
+	Length int64 `json:"length"`
+}
+
+// GenerateKey generates a new AES key.
+func (akgp *AesKeyGenParams) GenerateKey(
+	extractable bool,
+	keyUsages []CryptoKeyUsage,
+) (*CryptoKey[AesKeyAlgorithm], error) {
+	for _, usage := range keyUsages {
+		switch usage {
+		case WrapKeyCryptoKeyUsage, UnwrapKeyCryptoKeyUsage:
+			continue
+		case EncryptCryptoKeyUsage, DecryptCryptoKeyUsage:
+			if akgp.Algorithm.Name != AESKw {
```

Comment:
Could you add a comment here explaining why we're excluding AES-KW?

Should the error message be more informative in that case?

Reply:
Seems that it isn't supported by the stdlib https://github.com/golang/go/issues/27599

Reply:
I think I added one in another place, but will do it here too 👍🏻 

As Mihail pointed out, the reason is really that there is no official support for it from Go. In the previous implementation I wrote one myself, but didn't really trusted it.